### PR TITLE
Remove config_for as it is present in Rails

### PR DESCRIPTION
**Pull request by hallelujah**
It is already present in 4.2

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -34,10 +34,15 @@ def self.included(base)
   mattr_accessor :redis
 
   class Application < Rails::Application
+    def config_for(*args)
+      config = super
+      return config unless config.is_a?(Hash)
```
Comment (Martouta) [inline config/application.rb]: This looks easier to me:
```ruby
def config_for(*args)
  config = super
  config.is_a?(Hash) ? config.with_indifferent_access : config
end
```

But it is ok anyway :+1:

File: config/application.rb [hunk 2]
```diff
@@ -34,10 +34,15 @@ def self.included(base)
...
+      config.with_indifferent_access
+    end
 
     def simple_try_config_for(*args)
       config_for(*args)
-    rescue ConfigFor::ReadError, ConfigFor::MissingEnvironmentError => error
+    rescue => error
```
Comment (Martouta) [inline config/application.rb]: Does this mean the same as `rescue Exception => error` ? In that case:
1. I think it is better to specify it this way to it is clearer, but this is not the biggest issue about this.
2. @hallelujah Told me once that the problem of catching everything is that is literally eveything, including doing a `CTRL + C`, so I am not sure this is a very good idea :thinking:

File: config/application.rb (hunk 1 above)
Comment (Martouta) [reply config/application.rb]: Or even:
```ruby
def config_for(*args)
  config = super
  return config.with_indifferent_access if config.is_a?(Hash)
  config
end
```

File: config/application.rb (hunk 2 above)
Comment (hallelujah) [reply config/application.rb]: Not really you cannot rescue it see how Rails config_for is done.

**Pull request merged**